app/indexers/human_readable_type_indexer.rb

```diff
@@ -8,11 +13,28 @@ def initialize(resource:)
   def to_solr
     return {} unless decorated_resource.try(:human_readable_type)
     {
-      human_readable_type_ssim: decorated_resource.human_readable_type
+      human_readable_type_ssim: human_readable_type
     }
   end
 
   def decorated_resource
     @decorated_resource ||= resource.decorate
   end
+
+  private
+
+    def index_change_set?(change_set)
+      change_set_class_name = "#{change_set.camelize}ChangeSet"
+      change_set_class = change_set_class_name.constantize
+      self.class.change_sets.include?(change_set_class)
+    rescue NameError
+      Valkyrie.logger.warn("#{change_set} is not a valid resource type.")
+      false
+    end
+
+    def human_readable_type
```

Comment:
I'm not quite certain about this approach, but I felt that it was more acceptable to attempt something similar to this than to create an additional Indexer which just overrides the `human_readable_type_ssim` field values.